test/lib/generate/configs-test.ts

```diff
@@ -488,4 +488,55 @@ describe('generate (configs)', function () {
       expect(readFileSync('docs/rules/no-foo.md', 'utf8')).toMatchSnapshot();
     });
   });
+  describe('config as flat config', () => {
+    beforeEach(function () {
+      mockFs({
+        'package.json': JSON.stringify({
+          name: 'eslint-plugin-test',
+          exports: 'index.js',
+          type: 'module',
+        }),
+
+        'index.js': `
+          export default {
+            rules: { 'no-foo': { meta: { docs: { description: 'Description.' }, }, create(context) {} }, },
+            configs: {
+              recommended: [
+                {
+                  rules: {
+                    'test/no-foo': 'error',
+                  }
+                },
+                {
+                  rules: {
+                    'test/no-foo': 'error',
```

Comment:
Seems like we should have a separate rule `no-bar` in the second config object so that we can be sure that the rules from both objects were detected.